crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -266,9 +266,32 @@ impl Settlement {
         eth: &Ethereum,
         auction: &competition::Auction,
         revert_protection: &mempools::RevertProtection,
-    ) -> Result<competition::Score, boundary::Error> {
-        self.boundary
-            .score(eth, auction, self.gas.estimate, revert_protection)
+    ) -> Result<competition::Score, score::Error> {
+        let objective_value = self
+            .boundary
+            .objective_value(eth, auction, self.gas.estimate)?;
+
+        let score = match self.boundary.score() {
+            competition::SolverScore::Solver(score) => competition::Score(score),
+            competition::SolverScore::RiskAdjusted(success_probability) => competition::Score::new(
+                auction.score_cap(),
+                revert_protection,
+                objective_value,
+                self.gas.estimate,
+                auction.gas_price(),
+                success_probability,
+            )?,
+        };
```

Comment:
AFAIK the `domain` interfaces should already look like what they are supposed to look like if we didn't have any legacy code base. What I mean is that it looks like we simply pass whatever types the legacy code base needs into the `domain` type without considering a better design for it.
Having so many arguments already seems like a code smell to me. Maybe they could be packaged up in a nicer way?
I have no great idea which I find super convincing myself (still too little experience with DDD) but it seems to me that conceptually it should be enough to pass in the auction, the solution possibly `Ethereum` and maybe the revert protection.
This gets a bit awkward because we currently already have to compute the objective value to compare the score against in case the solver provided it but I'm not sure that should mean the signature of `Score::new()` has to be so cluttered.
As I mentioned I don't really have a great suggestion from the top of my head but wanted to bring it up so we don't end up with passing around lots and lots of arguments again like in the old code base.

Reply:
I can simplify this by merging `self.gas.estimate`, `self.gas.estimate` and `revert_protection` into a single parameter `cost_fail`, but can't do it before `solver` crate is dropped. Once we activate colocation this can be done.
I don't find this a final solution but rather a move forward.